x-pack/elastic-agent/pkg/agent/application/fleet_server_bootstrap.go

```diff
@@ -85,6 +86,11 @@ func newFleetServerBootstrap(
 
 	reporter := reporting.NewReporter(bootstrapApp.bgContext, log, bootstrapApp.agentInfo, logR)
 
+	if cfg.Settings.MonitoringConfig != nil {
+		cfg.Settings.MonitoringConfig.Enabled = false
+	} else {
+		cfg.Settings.MonitoringConfig = &monitoringCfg.MonitoringConfig{Enabled: false}
+	}
 	monitor, err := monitoring.NewMonitor(cfg.Settings)
```

Comment:
I attempted to replace the monitor with the noop monitor, but the agent was not able to bootstap correctly.